Design/Instrument-2.0/ScanningInstruments_ILL.md

```diff
@@ -0,0 +1,122 @@
+# Scanning Instruments for the ILL
+
+## Introduction
+
+Some details on requirements for the ILL were originally provided in a document from [development workshop discussions in 2015](https://github.com/mantidproject/documents/blob/master/Design/HandlingMovingInstruments.md).
+
+The three instruments concerned with this work at ILL are D2B, D4 and D7. D16 also potentially has a similar mode to D4, and the TAS instruments should be kept in mind, although not in the scope of this work.
+
+**Note:** These are some initial notes on the ILL instrument requirements. Some further validation and clarification of the requirements is still required.
+
+## Requirements
+
+### Mandatory
+* A loader can load data from a single file, and create the workspace with the correct positions and rotations stored for each detector
+* Multiple workspaces can be merged, with different sets of time indexes, and positions and rotations stored correctly for each detector
+* Detector positions for a given time index can still be modified after loading
+* Detector movement at the ILL is limited to the case of rotation around sample position - but full positional changes should be allowed, for example in case the sample is off-centre
+* Flexible step scans are supported, times at a given position can differ, as can angular step size changes between steps
+* Each spectrum ultimately links to a detector or detector group and a time index for scanning instruments
+* A start and end time, as the full time and date, can be obtained for each time index
+* Extra metadata can be accessed for a given position in the step scan - for example so normalisation to monitors can be performed later
+* Metadata values that change in time can be shared appropriately between positions in the step scan (for example in D2B files each file contains a single value for monitor counts, but 25 different detector positions, so the monitor counts for each position might need to be divided by 25)
+* Overlaps in detector positions are possible
+* The instrument view shows every step position for each detector, and deals with overlapping detectors
+* Saving/loading workspaces should deal with all of the information related to step scans
+
+### Desirable
+* Multiple workspaces can be merged with the same detector positions for certain time indexes - the data for spectra is then summed
+* Masking of detectors at different time indexes
```

Comment:
As mentioned above, I would handle this by two-level masking: Masking at the spectrum level and masking at the detector level.